test/excludes/ActiveRecord/ConnectionAdapters/PostgreSQLAdapter/QuotingTest.rb

```diff
@@ -0,0 +1,3 @@
+exclude :test_do_not_raise_when_int_is_not_wider_than_64bit, "replaced for correct quoting"
+exclude :test_do_not_raise_when_raise_int_wider_than_64bit_is_false, "replaced for correct quoting"
+exclude :test_raise_when_int_is_wider_than_64bit, "since integers are stringified there is not 64bit limit"
```

Comment:
I have to do some more documentation for this. Maybe @rafiss you know where to dig?

My current understanding of the context is that in ActiveRecord it seems that there is a 64bit cap for integer with the pg adapter. But since ints are stringified in crdb, there is no such thing for us.

Reply:
i'm surprised - it shouldn't be possible to have an integer wider than 64 bits in CRDB.

Reply:
Ok I'll dig further then

Reply:
So what I understood from this test: when quoting in pg, activerecord checks if the int is withing the 64 bit range with the `#check_int_in_range` method. The error message related to that is:

```ruby
        def check_int_in_range(value)
          if value.to_int > 9223372036854775807 || value.to_int < -9223372036854775808
            exception = <<~ERROR
              Provided value outside of the range of a signed 64bit integer.


              PostgreSQL will treat the column type in question as a numeric.
              This may result in a slow sequential scan due to a comparison
              being performed between an integer or bigint value and a numeric value.


              To allow for this potentially unwanted behavior, set
              ActiveRecord.raise_int_wider_than_64bit to false.
            ERROR
            raise IntegerOutOf64BitRange.new exception
          end
        end
```

https://github.com/rails/rails/blob/4e01a7e8753c30e22264e185082524a9ea5e0c90/activerecord/lib/active_record/connection_adapters/postgresql/quoting.rb#L28-L42

So PG would think of strings containing only numbers as int, and would be way slower if they exceed 64bit. @rafiss does this ring a bell to you? Is there a similar behaviour in cockroach?

I'll keep looking for documentation related to that in pg to be sure. Might edit this message when I find an answer.

The fix, if need be, is fairly easy.

Reply:
ah i understand now - so if it's larger than 64 bits it's treated as `decimal` (also called `numeric`) rather than `integer`

i think matching PG here makes sense. it's fine to make that change later too.

Reply:
thanks for the nice research on this!

Reply:
@rafiss sorry didn't see your message before editing. I think this needs our attention!

Reply:
After investigating some more, this adapter is not affected by the [mentioned vulnerability](https://github.com/advisories/GHSA-579w-22j4-4749). When quoting a number using this gem, it will be converted to a string first, taking advantage of cockroach's behavior of converting implicitly strings into numbers when needed. That conversion done on the server side is checking for bounds and raising if need be.

Here's an example of some code that could be abused in the pg adapter before the patch, but that raises in this adapter:

```ruby
# age is an int column with an index
Cat.where("age = ?", 2**63)
# => PG::InvalidTextRepresentation: ERROR:  could not parse "9223372036854775808" as type int: strconv.ParseInt: parsing "9223372036854775808": value out of range (ActiveRecord::StatementInvalid)
```